flow_runner.go

```diff
@@ -74,52 +108,45 @@ func (f *flowRunner) Run(ctx context.Context, args []string) int { //nolint // T
 			usageRequested = true
 			return nil
 		}
-		fmt.Fprintf(f.output, "unknown argument: %s\n", arg)
 		return fmt.Errorf("unknown argument: %s", arg)
 	}
 
 	for _, arg := range args {
 		err := argHandler(arg)
 		if err != nil {
-			return CodeInvalidArgs
+			return []string{}, false, err
 		}
 	}
+	return tasks, usageRequested, nil
+}
 
-	if usageRequested {
-		printUsage(f)
-		return CodePass
-	}
-
-	// set default task if none is provided
-	if len(tasks) == 0 && f.defaultTask.name != "" {
-		tasks = append(tasks, f.defaultTask.name)
+func (f *flowRunner) defaultTasks(tasks []string) []string {
+	if len(tasks) > 0 || len(f.defaultTask.name) == 0 {
```

Comment:
why have you canged `f.defaultTask.name != "" ` to `len(f.defaultTask.name) == 0` ?
for me `!= ""` is more explicit, type-safe and readable

I know a few years ago the argument was that it resulted in more optimized code but it is no longer valid
reference: https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go

Reply:
My reasoning was not about optimized code, but rather testing in a unified way like slices (which shouldn't be tested against `nil` for example).
I've changed it back.